apps/src/lib/script-editor/FlexGroup.jsx

```diff
@@ -0,0 +1,190 @@
+    borderTopLeftRadius: borderRadius,
+    borderTopRightRadius: borderRadius,
+    padding: 10
+  },
+  groupBody: {
+    background: color.lightest_cyan,
+    borderBottomLeftRadius: borderRadius,
+    borderBottomRightRadius: borderRadius,
+    padding: 10,
+    marginBottom: 20
+  },
+  addGroup: {
+    fontSize: 14,
+    color: 'white',
+    background: color.cyan,
+    border: `1px solid ${color.cyan}`,
+    boxShadow: 'none',
+    margin: '0 0 30px 0'
+  },
+  addStage: {
+    fontSize: 14,
+    color: '#5b6770',
+    background: 'white',
+    border: '1px solid #ccc',
+    boxShadow: 'none',
+    margin: '0 10px 10px 10px'
+  }
+};
+
+const FlexGroup = React.createClass({
+  propTypes: {
+    addGroup: React.PropTypes.func.isRequired,
+    addStage: React.PropTypes.func.isRequired,
+    stages: React.PropTypes.array.isRequired
+  },
+
+  handleAddGroup() {
+    this.props.addGroup(prompt('Enter new stage name'), prompt('Enter new group name'));
+  },
+
+  handleAddStage(position) {
+    this.props.addStage(position, prompt('Enter new stage name'));
+  },
+
+  serializeStages(stages) {
+    let s = [];
+    stages.forEach(stage => {
+      let t = `stage '${stage.name}'`;
+      if (stage.lockable) {
+        t += ', lockable: true';
+      }
+      if (stage.flex_category) {
+        t += `, flex_category: '${stage.flex_category}'`;
+      }
+      s.push(t);
+      stage.levels.forEach(level => {
+        if (level.ids.length > 1) {
+          s.push('variants');
+          level.ids.forEach(id => {
+            s = s.concat(this.serializeLevel(id, level, level.activeId === id));
+          });
+          s.push('endvariants');
+        } else {
+          s = s.concat(this.serializeLevel(level.ids[0], level));
+        }
+      });
+      s.push('');
+    });
+    return s.join('\n');
+  },
+
+  serializeLevel(id, level, active) {
+    if (id === NEW_LEVEL_ID) {
+      return;
+    }
+
+    const s = [];
+    const key = this.props.levelKeyList[id];
+    if (/^blockly:/.test(key)) {
+      if (level.skin) {
+        s.push(`skin '${level.skin}'`);
+      }
+      if (level.videoKey) {
+        s.push(`video_key_for_next_level '${level.videoKey}'`);
+      }
+      if (level.concepts) {
+        s.push(`concepts ${level.concepts}`);
+      }
+      if (level.conceptDifficulty) {
+        s.push(`level_concept_difficulty '${level.conceptDifficulty}'`);
+      }
+    }
+    let l = `${this.normalizeLevelKind(level.kind)} '${key.replace(/'/, "\\'")}'`;
+    if (active === false) {
+      l += ', active: false'
+    }
+    s.push(l);
+    return s;
+  },
+
+  normalizeLevelKind(kind) {
+    return (!kind || kind === 'puzzle' || kind === 'unplugged') ? 'level' : kind;
+  },
+
+  render() {
+    const groups = _.groupBy(this.props.stages, stage => (stage.flex_category || 'Default'));
+    let count = 0;
+    let afterStage = 1;
+
+    return (
+      <div>
+        {_.map(groups, (stages, group) => {
+          return (
```

Comment:
nit, I think you could reduce indentation if you wanted by getting rid of the return statement and doing:
```
{_.map(groups, (stages, group) => (
  <div key={group}>
    ...
  </div>
)}